app/code/Magento/Catalog/Model/Product/Image/Cache.php

```diff
@@ -127,4 +154,97 @@ protected function processImageData(Product $product, array $imageData, $file)
 
         return $this;
     }
+
+    /**
+     * Get themes in use
+     *
+     * @return array
+     */
+    private function getThemesInUse(): array
+    {
+        $themesInUse = [];
+
+        $registeredThemes = $this->themeCollection->loadRegisteredThemes();
+        $storesByThemes   = $this->themeCustomizationConfig->getStoresByThemes();
+
+        foreach ($registeredThemes as $registeredTheme) {
+            if (array_key_exists($registeredTheme->getThemeId(), $storesByThemes)) {
+                $themesInUse[] = $registeredTheme;
+            }
+        }
+
+        $connection = $this->attribute->getResource()->getConnection();
+
+        $productEntityTypeId = \Magento\Catalog\Setup\CategorySetup::CATALOG_PRODUCT_ENTITY_TYPE_ID;
+
+        $productCustomDesignAttributeId = $this->attribute->loadByCode($productEntityTypeId, 'custom_design')->getId();
```

Comment:
Please use \Magento\Eav\Model\Config::getAttribute instead of load entity by model